sources/core/Stride.Core.CompilerServices/Analyzers/STRDIAG007DataMemberOnDelegate.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Stride.Core.CompilerServices.Common;
+
+namespace Stride.Core.CompilerServices.Analyzers;
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal class STRDIAG007DataMemberOnDelegate : DiagnosticAnalyzer
+{
+    public const string DiagnosticId = "STRDIAG007";
+    private const string Title = "Invalid DataMemberAttribute";
+    private const string MessageFormat = "Invalid DataMembermode for the specified member '{0}'. A Delegate can't be a Delegate.";
+    private const string Category = "CompilerServices";
+
+    private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+        DiagnosticId,
+        Title,
+        MessageFormat,
+        Category,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.EnableConcurrentExecution();
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+        context.RegisterCompilationStartAction(AnalyzeCompilationStart);
+    }
+    private static void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+    {
+        var dataMemberAttribute = WellKnownReferences.DataMemberAttribute(context.Compilation);
+        if (dataMemberAttribute is null)
+        {
+            return;
+        }
+
+        context.RegisterSymbolAction(symbolContext => AnalyzeProperty(symbolContext, dataMemberAttribute), SymbolKind.Property);
+        context.RegisterSymbolAction(symbolContext => AnalyzeField(symbolContext, dataMemberAttribute), SymbolKind.Field);
+    }
+    private static void AnalyzeField(SymbolAnalysisContext context, INamedTypeSymbol dataMemberAttribute)
+    {
+        var fieldSymbol = (IFieldSymbol)context.Symbol;
+
+        if (!WellKnownReferences.HasAttribute(fieldSymbol, dataMemberAttribute))
+            return;
+
+        var fieldType = fieldSymbol.Type;
+
+        if (fieldType is null)
+            return;
+
+        if (fieldType.TypeKind == TypeKind.Delegate)
+        {
+            DiagnosticsAnalyzerExtensions.ReportDiagnostics(Rule, context, dataMemberAttribute, fieldSymbol);
+        }
+    }
+
+    private static void AnalyzeProperty(SymbolAnalysisContext context, INamedTypeSymbol dataMemberAttribute)
+    {
+        var propertySymbol = (IPropertySymbol)context.Symbol;
+
+        if (!WellKnownReferences.HasAttribute(propertySymbol, dataMemberAttribute))
+            return;
+        var propertyType = propertySymbol.Type;
+
+        if (propertyType is null) 
+            return;
+
+        if(propertyType.SpecialType == SpecialType.System_Delegate)
```

Comment:
The check in the field method is correct - here you only check for the exact `System.Delegate` type instead of any such `Action`, `Func<>` etc.

Reply:
[cff1c97](https://github.com/stride3d/stride/pull/1864/commits/cff1c9797d1d0cc579a1c853a5bd9973f950c979)

this is fixed now.
Forgot to fix it after i applied it to field.